pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -953,6 +953,41 @@ func ValidateVirtualMachineInstanceSpec(field *k8sfield.Path, spec *v1.VirtualMa
 		})
 	}
 
+	if spec.Domain.Devices.HostDevices != nil && !config.HostDevicesPassthroughEnabled() {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("Host Devices feature gate is not enabled in kubevirt-config"),
+			Field:   field.Child("HostDevices").String(),
+		})
+	}
+	if hostDevs := config.GetPermittedHostDevices(); hostDevs != nil {
+		// build a map of all permitted host devices
+		supportedHostDevicesMap := make(map[string]bool)
+		for _, dev := range hostDevs.PciHostDevices {
+			supportedHostDevicesMap[dev.ResourceName] = true
+		}
+		for _, dev := range hostDevs.MediatedDevices {
+			supportedHostDevicesMap[dev.ResourceName] = true
+		}
+		for _, hostDev := range spec.Domain.Devices.GPUs {
+			if _, exist := supportedHostDevicesMap[hostDev.DeviceName]; !exist {
+				causes = append(causes, metav1.StatusCause{
+					Type:    metav1.CauseTypeFieldValueInvalid,
+					Message: fmt.Sprintf("GPU %s is not permitted in kubevirt-host-device-plugin-config", hostDev.DeviceName),
+					Field:   field.Child("GPUs").String(),
```

Comment:
How would we go about this? Yes, they won't be able to start *new* VMs the GPU they are using is not permitted, but it won't affect the running ones, am I correct?

Reply:
Running VMs wouldn't be able to be shutdown because the validation webhook would reject altering the VM spec's runStratery due to the VMI template failing validation. 

So running VMs won't immediately get terminated, but from what I see it will impact the ability to manage existing VMs. 

> How would we go about this? 

maybe we can't use the permit list for the existing GPUs... or maybe if a permit list doesn't exist that should mean GPUs devices aren't validated.

Reply:
oh I see. 

I think device validation/whitelisting is one of the key points of this PR. I wouldn't want to lose the ability to validate GPUs...

Can't we get away with adding documentation, explaining that in order to shut down a long-running VMI with a GPU you have to whitelist it first?

Reply:
> Can't we get away with adding documentation, explaining that in order to shut down a long-running VMI with a GPU you have to whitelist it first?

I'm not sure if it's just shutting down the vm. There could be other implications, that's just one that came to mind.

Here's the idea i was working with. it's a little odd but it would preserve backwards compatibility. PermittedHostDevices is a pointer

```
PermittedHostDevices        *PermittedHostDevices 
```

Since it's a pointer we can treat that as a boolean of sorts. For GPUs, could we say that the whitelist is only applied at validation time if the PermittedHostDevices struct is set (not nil), even if it's set with empty lists. This would allow people to opt into the new validation behavior for GPUs. Existing deployments would simply work.

The new VMI HostDevices on the other hand would always require validation against the PermittedHostDevices since we know no one is using that new field yet. 

It's not exactly an elegant solution to the problem, but maybe it would work?

Reply:
Yup, it's a good idea.
I was previously concerned and didn't do anything like this since I thought we are providing a way to skip validation of permitted devices, but on the other hand, it's something that's controlled by the admin anyway.

Alright, thanks! let me add this quickly.

Reply:
@davidvossel 
oh... it's already like that on line 975, `if hostDevs := config.GetPermittedHostDevices(); hostDevs != nil { `

Reply:
I've added a unit test to cover this.

Reply:
ha, i missed that, sounds like there's no issue then